From a commit ???
From: Shadow
Date: Mon, Apr 4 08:50:33 2022
Subject: [PATCH] Fixed the PDE_DATA calls for newer kernels.

Tentative patch for broadcom-sta 6.30.223.271 driver for Linux 5.17

Applies on top of all the patches applied to broadcom-wl-dkms 6.30.223.271-23 on Arch Linux.

Signed-off-by: Unknown
---
 src/wl/sys/wl_linux.c           | 2 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -3244,7 +3244,7 @@
 static ssize_t
 wl_proc_read(struct file *filp, char __user *buffer, size_t length, loff_t *offp)
 {
-	wl_info_t * wl = PDE_DATA(file_inode(filp));
+	wl_info_t * wl = pde_data(file_inode(filp));
 #endif
 	int bcmerror, len;
 	int to_user = 0;
@@ -3301,7 +3301,7 @@
 static ssize_t
 wl_proc_write(struct file *filp, const char __user *buff, size_t length, loff_t *offp)
 {
-	wl_info_t * wl = PDE_DATA(file_inode(filp));
+	wl_info_t * wl = pde_data(file_inode(filp));
 #endif
 	int from_user = 0;
 	int bcmerror;